### PR TITLE
feat: add `fs/read-ndjson`

### LINKED ISSUE #1074: [RFC]: add `@stdlib/fs/read-ndjson`
### Description

This RFC proposes adding a filesystem utility to read a file which is newline-delimited JSON (NDJSON).

Package: `@stdlib/fs/read-ndjson`
Alias: `readNDJSON`

### Related Issues

The following RFC is a prerequisite to being able to implement this feature:

- https://github.com/stdlib-js/stdlib/issues/1075

### Questions

No.

### Other

For a reference package, see

- [`@stdlib/fs/read-json`](https://github.com/stdlib-js/stdlib/tree/1dbb5491b16d4f969cf4748bd1fc7bcd80b89ccc/lib/node_modules/%40stdlib/fs/read-json)

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by yaswanthkosuru**
Resolves #1074.

## Description

his RFC proposes adding a filesystem utility to read a file which is newline-delimited JSON (NDJSON).

## Related Issues

[https://github.com/stdlib-js/stdlib/issues/1075](https://github.com/stdlib-js/stdlib/issues/1075)

This pull request:

-   resolves https://github.com/stdlib-js/stdlib/issues/1074

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by yaswanthkosuru** [comment]
@Planeshifter  ready for initial review!

File: lib/node_modules/@stdlib/fs/read-ndjson/README.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# Read JSON
```
Comment (kgryte) [inline lib/node_modules/@stdlib/fs/read-ndjson/README.md]:
```suggestion
# readNDJSON
```

File: lib/node_modules/@stdlib/fs/read-ndjson/README.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+> Read a file as [JSON][json].
```
Comment (kgryte) [inline lib/node_modules/@stdlib/fs/read-ndjson/README.md]: These descriptions need updating.

Comment (kgryte) [reply lib/node_modules/@stdlib/fs/read-ndjson/README.md]: Should be 
```suggestion
> Read a file as [newline-delimited JSON][ndjson].
```

Similar changes should be made throughout this package.

File: lib/node_modules/@stdlib/_tools/lint/filenames/lib/extensions.json [hunk 3]
```diff
@@ -45,5 +45,6 @@
   ".wat",
   ".woff",
   ".yml",
-  ".xml"
+  ".xml",
+  ".ndjson"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/_tools/lint/filenames/lib/extensions.json]: This should be inserted in alphabetical order.

**Comment by yaswanthkosuru** [comment]
Should I add these link as ndjson :https://ndjson.org/ in readme. md
I am not sure?

**Comment by kgryte** [comment]
Yes, that would be preferrable.

**Comment by yaswanthkosuru** [comment]
@kgryte  Ready for final review

**Comment by yaswanthkosuru** [comment]
@Planeshifter  it has been almost 3 weeks.i am free now, can you please review my pr
Thanks ..

**Review by kgryte — APPROVED** [review]
This now LGTM. Thanks for your contribution, @yaswanthkosuru.

**Pull request merged**